src/Shared/ExceptionHandling.cs

```diff
@@ -157,7 +156,9 @@ internal static bool IsIoRelatedException(Exception e)
             return e is UnauthorizedAccessException
                    || e is NotSupportedException
                    || (e is ArgumentException && !(e is ArgumentNullException))
-                   || e is SecurityException
+#if NETFRAMEWORK
+                   || e is System.Security.SecurityException
+#endif
```

Comment:
I don't think you have to conditionalize this one; it appears to work fine in .NET Core.

```suggestion
                   || e is System.Security.SecurityException
```

(you could leave System.Security in the usings and leave this part entirely unchanged, probably)